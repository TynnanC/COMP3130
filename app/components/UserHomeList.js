import React from 'react';
import { View, StyleSheet, Image, Text } from 'react-native';

import AppColors from '../config/AppColors';

function UserHomeList({ profileImage, firstName, lastName, username, email }) {
    //Returns the list that displays the users details.
    return (
        <>
            <View style={styles.image}>
                {isFinite(profileImage) ? <Image source={profileImage} style={styles.profileImage} /> : <Image source={{ uri: profileImage }} style={styles.profileImage} />}
            </View>
            <View style={styles.main}>
                <Text style={styles.text}>Welcome {username}!</Text>
            </View>
            <View style={styles.sub}>
                <Text><Text style={styles.textEffect}>First Name:</Text> {firstName}</Text>
            </View>
            <View style={styles.sub}>
                <Text><Text style={styles.textEffect}>Last Name:</Text> {lastName}</Text>
            </View>
            <View style={styles.sub}>
                <Text><Text style={styles.textEffect}>Email:</Text> {email}</Text>
            </View>
        </>
    );
}

const styles = StyleSheet.create({
    profileImage: {
        width: 150,
        height: 150,
        borderRadius: 5,
        marginTop: '5%'
    },
    image: {
        flexDirection: "column",
        alignItems: "center",
        marginBottom: '5%'
    },
    main: {
        flexDirection: "column",
        alignItems: "center",
        justifyContent: 'center',
        backgroundColor: AppColors.SecondaryDull,
        borderRadius: 20,
        marginBottom: '10%',
        padding: 10,
    },
    text: {
        fontWeight: 'bold'
    },
    sub: {
        flexDirection: "row",
        justifyContent: 'flex-start',
        alignItems: "center",
        backgroundColor: AppColors.Grey,
        borderRadius: 20,
        marginBottom: '10%',
        padding: 10,
    },
    textEffect: {
        fontWeight: 'bold'
    }
});

export default UserHomeList;