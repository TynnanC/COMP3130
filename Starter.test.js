import React from 'react';
import renderer from 'react-test-renderer';
import UserHomeList from './app/components/UserHomeList.js';

test('UserHomeList renders correctly', () => {
    const tree = renderer.create(<UserHomeList />).toJSON();
    expect(tree).toMatchSnapshot();
});