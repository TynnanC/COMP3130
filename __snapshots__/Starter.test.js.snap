// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHomeList renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "marginBottom": "5%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 5,
          "height": 150,
          "marginTop": "5%",
          "width": 150,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff2e6",
        "borderRadius": 20,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginBottom": "10%",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Welcome 
      !
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": "10%",
        "padding": 10,
      }
    }
  >
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        First Name:
      </Text>
       
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": "10%",
        "padding": 10,
      }
    }
  >
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Last Name:
      </Text>
       
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": "10%",
        "padding": 10,
      }
    }
  >
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Email:
      </Text>
       
    </Text>
  </View>,
]
`;
